Bigtable/src/ResumableStream.php

```diff
@@ -175,11 +175,11 @@ public function readAll()
                         $currentAttempt = 0; // reset delay and attempt on successful read.
                     }
                 } catch (\Exception $ex) {
-                    $totalAttempt++;
-                    $currentAttempt++;
                 }
+                $totalAttempt++;
+                $currentAttempt++;
             }
-        } while ((!$this->retryFunction || ($this->retryFunction)($ex)) && $currentAttempt <= $this->retries);
+        } while (($this->retryFunction)($ex) && $currentAttempt <= $this->retries);
```

Comment:
not sure if this is idiomatic for php, but in java the ctor would check that the passed in retryFunction is non-null so that if it was, the stack trace would include the responsible party

Reply:
in this case, passing in `null` as the retry function (in `ResumableStream`'s constructor) would throw a PHP type error, as the parameter is not nullable